packages/orchestration/src/utils/address.js

```diff
@@ -141,3 +142,42 @@ export const parseAccountId = partialId => {
   throw Fail`Invalid accountId: ${q(partialId)}`;
 };
 harden(parseAccountId);
+
+/**
+ * Extracts the namespace from a fully-qualified CAIP-10 address.
+ *
+ * @param {AccountId} address - The CAIP-10 address.
+ * @returns {string} - The namespace.
+ */
+export const getNamespace = address => {
```

Comment:
Why these three new functions instead of using `parseAccountId` ?

Reply:
I couldn't get the types to be sure that `.namespace` and `.reference` were known to be present in the output of `parseAccountId()`. `parseAccountId` doesn't know whether the parameter is fully qualified.

Reply:
`resolveAccountId()` now guarantees that its result is CAIP-10, so it can be supplied to `parseAccountId()`.

It seems circuitous to me that we use the former to build a string that we then parse, rather than writing a tool that can take any of our address formats, and return an object that can return any of the needed parts, or produce whichever string format is useful. (CosmosChainAddress or CAIP-10 for instance, or whatever Solana will want.)

But if we want CAIP-10 to be our canonical representation, I suppose splitting strings when we want the parts back is the cleanest choice.

Reply:
> It seems circuitous to me that we use the former to build a string that we then parse, rather than writing a tool that can take any of our address formats

That would be and it's not what I proposed. I said for the case of "any of our address formats" to have a looser function, `parseAccountIdArg`. (Note the "Arg")

Reply:
The extra transformations cost cycles but as you conclude, having a single canonical representation is worth it.